### PR TITLE
pango: Trust upstream nullable annotations for return values

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #789.

### CODE REVIEW

File: pango/Gir.toml [hunk 1]
```diff
@@ -9,6 +9,7 @@ min_cfg_version = "1.40"
 use_gi_docgen = true
 single_version_file = true
 deprecate_by_min_version = true
+trust_return_value_nullability = true
```
Comment (sdroege) [inline pango/Gir.toml:12]: Also remove all the `nullable` overrides from this file then :)

Comment (GuillaumeGomez) [reply pango/Gir.toml:12]: Ok!

**Comment by GuillaumeGomez** [comment]
Updated!

File: pango/src/auto/context.rs [hunk 2]
```diff
@@ -30,7 +30,7 @@ glib::wrapper! {
 
 impl Context {
     #[doc(alias = "pango_context_new")]
-    pub fn new() -> Context {
+    pub fn new() -> Option<Context> {
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply pango/src/auto/context.rs:33]: https://gitlab.gnome.org/GNOME/pango/-/merge_requests/651

File: pango/src/auto/context.rs [hunk 3]
```diff
@@ -83,8 +83,8 @@ impl Context {
 
     #[doc(alias = "pango_context_get_language")]
     #[doc(alias = "get_language")]
-    pub fn language(&self) -> Option<Language> {
-        unsafe { from_glib_full(ffi::pango_context_get_language(self.to_glib_none().0)) }
+    pub fn language(&self) -> Language {
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply pango/src/auto/context.rs:86]: I don't think it's possible: https://gitlab.gnome.org/GuillaumeGomez/pango/-/blob/main/pango/pango-context.c#L70

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply pango/src/auto/context.rs:86]: And when you set the language too: https://gitlab.gnome.org/GuillaumeGomez/pango/-/blob/main/pango/pango-context.c#L400

[the remark this answers is not in the document]
Comment (sdroege) [reply pango/src/auto/context.rs:86]: That's why I said it looks like a bug :) Look at what the setter is actually returning: https://gitlab.gnome.org/GuillaumeGomez/pango/-/blob/main/pango/pango-context.c#L416

`set_language` is only set by `pango_context_set_language()` to the value passed in, and that can be `NULL`. It is also initialized to `NULL` on construction (implicitly).

[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply pango/src/auto/context.rs:86]: Oh indeed.

[the remark this answers is not in the document]
Comment (sdroege) [reply pango/src/auto/context.rs:86]: I would keep it as nullable here for now and report, separately, an issue against pango to maybe have this function fixed if that's indeed unintentionally strange.

**Comment by sdroege** [comment]
For moving forward here, I would suggest you add overrides for the wrong nullable cases to `Gir.toml`. Then the API here is fixed at least and we can get rid of the overrides once we have the new .gir file

File: pango/src/auto/context.rs [hunk 4]
```diff
@@ -84,7 +84,7 @@ impl Context {
     #[doc(alias = "pango_context_get_language")]
     #[doc(alias = "get_language")]
     pub fn language(&self) -> Option<Language> {
-        unsafe { from_glib_full(ffi::pango_context_get_language(self.to_glib_none().0)) }
+        unsafe { from_glib_none(ffi::pango_context_get_language(self.to_glib_none().0)) }
     }
 
     #[doc(alias = "pango_context_get_matrix")]
```
Comment (sdroege) [inline pango/src/auto/context.rs:90]: Further below, `fn metrics()` should return no `Option` according to your pango MR

File: pango/Gir.toml [hunk 5]
```diff
@@ -206,6 +203,14 @@ boxed_inline = true
 name = "Pango.Context"
 status = "generate"
     [[object.function]]
+    name = "new"
+        [object.function.return]
+        nullable = false
+    [[object.function]]
+    name = "get_language"
+        [object.function.return]
+        nullable = true
```
Comment (sdroege) [inline pango/Gir.toml:212]: Add links here to your MR that makes the overrides unnecessary

Comment (GuillaumeGomez) [reply pango/Gir.toml:212]: Good idea!

File: pango/src/auto/context.rs (hunk 4 above)
Comment (GuillaumeGomez) [reply pango/src/auto/context.rs:90]: Indeed, fixed.

**Comment by sdroege** [comment]
regen check is unhappy

File: pango/Gir.toml [hunk 6]
```diff
@@ -206,10 +203,26 @@ boxed_inline = true
...
 status = "generate"
     [[object.function]]
+    name = "new"
+        # Can be removed once https://gitlab.gnome.org/GNOME/pango/-/merge_requests/651
+        # has been merged.
+        [object.function.return]
+        nullable = false
+    [[object.function]]
+    name = "get_language"
+        # Can be removed once https://gitlab.gnome.org/GNOME/pango/-/merge_requests/651
```
Comment (sdroege) [inline pango/Gir.toml:213]: You don't change that one there. I think that needs a separate issue to first of all figure out how this function is supposed to work.

Comment (GuillaumeGomez) [reply pango/Gir.toml]: Done.

File: glib/src/auto/enums.rs [hunk 7]
```diff
@@ -1346,6 +1346,10 @@ pub enum UnicodeScript {
     #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_72")))]
     #[doc(alias = "G_UNICODE_SCRIPT_MATH")]
     Math,
+    #[doc(alias = "G_UNICODE_SCRIPT_KAWI")]
+    Kawi,
+    #[doc(alias = "G_UNICODE_SCRIPT_NAG_MUNDARI")]
+    NagMundari,
```
Comment (sdroege) [inline glib/src/auto/enums.rs:1352]: These two needs to be marked as `version = 2.74` in the `Gir.toml`

**Review by sdroege — APPROVED** [review]

**Pull request merged**